### ISSUE TITLE
Embed Camunda WebApp

### ISSUE THREAD

**Issue by hawky-4s-** [camunda/camunda-bpm-spring-boot-starter#3]
https://github.com/Zapata/camunda-spring-boot-embedded is embedding the camunda-webapp as a web-jar. Maybe we can do the same? WDYT?

**Comment by osteinhauer** [comment]
The trick is that in gradle build the war ist repackaged as webjar (https://github.com/Zapata/camunda-spring-boot-embedded/blob/master/camunda-webapp-webjar/build.gradle).

IMHO: A spring boot starter should never do such "hacks". And the example I have provided is very ugly too.

But the idea to have an webjar is great. We should ask if the webapp module can provided an webjar.

**Comment by hawky-4s-** [comment]
Valid point. I'll take care of that.

**Comment by osteinhauer** [comment]
Thanks for that. Until there is no idea to include the webapp in an nice way the starter should not be used by anyone.

**Comment by jangalinski** [comment]
Well, then this should not be part of the master branch anymore ... we should aim for a soon POC release so we can play around with the stack, and everything thats not supposed to work is just confusing ... so maybe create a feature branch ad put it on hold?

**Comment by osteinhauer** [comment]
feature branch is a good idea. We should do that after we have all agreed to the multi module.

**Comment by jangalinski** [comment]
Any news on the web jar deliverable from platform? Is it realistic to have this in 7.4? What needs to be done?

**Comment by meyerdan** [comment]
Pull request ?   :=)

**Comment by jangalinski** [comment]
I just created https://github.com/camunda/camunda-bpm-webapp/pull/5

Please check my remarks, I am not quite happy with the result, but cant figure out whats missing, I guess thats because I need some sleep ...

**Comment by jangalinski** [comment]
If you have time, please check the PR https://github.com/camunda/camunda-bpm-webapp/pull/5 comments .... Daniel is asking me if the generated output is what I need, and honestly: I do not know ... I suppose there is some stuff missing ...

**Comment by jangalinski** [comment]
Even when we have the webjar, this will only be applicable for 7.4, so I moved it to milestone 2.0

**Comment by jangalinski** [comment]
I started the following: provide a camunda-bpm-spring-boot-webjar inside this repo. This is exactly the same as the webjar PR for 7.4, but we generate it ourselves, so we can use it with 7.3 right away. for 7.4 this can be removed. 
I also locally created a c-b-s-b-starter-webapp where this webjar is used and am currently integrating the ServletFilter from the zapata approach (using Christians branch). Hope to get this running soon.

**Comment by abrass** [comment]
Would be great to have this feature. 

Is there a PR for spring boot starter for any camunda version, 7.3 or 7.4 including the cockpit as a web-jar?

greetz

Anton

**Comment by osteinhauer** [comment]
If you don't want to wait until Cockpit is supported you can have a look at https://github.com/camunda/camunda-bpm-spring-boot-starter/tree/Embed_Camunda_WebApp_%233/camunda-webapp-spring-boot-starter and do the configuration on your own. Be aware that this isn't nice at all. That is why it isn't integrated yet. See the discussion above.

**Comment by hawky-4s-** [comment]
Okay, I merged the PR on Camunda-Webapp

**Comment by osteinhauer** [comment]
The webjar is integrated now but not tested well. 

Beside the webjar isn't bundled very nice:

The resources should be located in ``META-INF/resources`` like http://www.webjars.org and https://spring.io/blog/2014/01/03/utilizing-webjars-in-spring-boot describes. Having that it wouldn't be necessary to add all resources from the classpath root like [CamundaBpmWebappAutoConfiguration.java](https://github.com/camunda/camunda-bpm-spring-boot-starter/blob/master/extension/camunda-spring-boot-starter-webapp/src/main/java/org/camunda/bpm/spring/boot/starter/webapp/CamundaBpmWebappAutoConfiguration.java) does: ``registry.addResourceHandler("/**").addResourceLocations("classpath:/")`` 

Maybe another one has an idea to get the resources served in a better way?

**Comment by osteinhauer** [comment]
This feature seems to work and will be contained in release 1.1.0
Bugs and features will go in separate issues

**Comment by jangalinski** [comment]
Question (cant test it myself right now): If I use starter-webapp and starter-rest together, can I still access the rest-api, or is that masked by the webapp resource?

**Comment by osteinhauer** [comment]
Just created an example (rest-api-and-webapp <https://github.com/osteinhauer/camunda-spring-boot-starter-examples/tree/master/rest-api-and-webapp>) because I was interested too. I figured out that both can be used in parallel.
